crates/biome_js_analyze/src/lint/nursery/no_useless_length_check.rs

```diff
@@ -0,0 +1,366 @@
+            let AnyJsExpression::JsIdentifierExpression(array_token) = comparing_zero else {
+                return None;
+            };
+            let key = array_token.text();
+            if let Some(comparing_zero_list) = comparing_zeros.get_mut(&key) {
+                comparing_zero_list.push(replacer?);
+            } else {
+                comparing_zeros.insert(key, vec![replacer?]);
+            }
+            Some(())
+        }
+        // .some() or .every() etc.
+        AnyJsExpression::JsCallExpression(task_exp) => {
+            if task_exp.is_optional_chain() {
+                return None;
+            }
+            let task_member_exp =
+                AnyJsMemberExpression::cast(task_exp.callee().ok()?.into_syntax())?;
+            let task_target = task_member_exp.object().ok()?;
+            let AnyJsExpression::JsIdentifierExpression(task_target_token) = task_target else {
+                return None;
+            };
+
+            let task_member_name_node = task_member_exp.member_name()?;
+            let task_member_name = task_member_name_node.text();
+            match expect_error {
+                ErrorType::UselessLengthCheckWithEvery => {
+                    if task_member_name == "every" {
+                        array_tokens_used_api.insert(task_target_token.text());
+                        Some(())
+                    } else {
+                        None
+                    }
+                }
+                ErrorType::UselessLengthCheckWithSome => {
+                    if task_member_name == "some" {
+                        array_tokens_used_api.insert(task_target_token.text());
+                        Some(())
+                    } else {
+                        None
+                    }
+                }
+            }
+        }
+        // ( foo )
+        AnyJsExpression::JsParenthesizedExpression(parent_exp) => search_logical_exp(
+            &parent_exp.expression().ok()?,
+            replacer,
+            expect_error,
+            comparing_zeros,
+            array_tokens_used_api,
+        ),
+        AnyJsExpression::JsIdentifierExpression(_) => Some(()),
+        _ => None,
+    }
+}
+
+fn get_parenthesized_parent(exp: AnyJsExpression) -> AnyJsExpression {
+    let Some(parent) = exp.syntax().parent() else {
+        return exp;
+    };
+    let Some(parent) = JsParenthesizedExpression::cast(parent) else {
+        return exp;
+    };
+    get_parenthesized_parent(AnyJsExpression::from(parent))
+}
+
+fn get_parenthesized_child(exp: &AnyJsExpression) -> AnyJsExpression {
+    let AnyJsExpression::JsParenthesizedExpression(parenthesized) = exp else {
+        return exp.clone();
+    };
+    let Some(child) = parenthesized.expression().ok() else {
+        return exp.clone();
+    };
+    get_parenthesized_child(&child)
+}
+
+impl Rule for NoUselessLengthCheck {
+    type Query = Ast<JsLogicalExpression>;
+    type State = (ErrorType, Replacer);
```

Comment:
I believe it's best to use a `struct` for the state, so you can document every field. As it is, it's difficult to understand what is it.